src/client/Microsoft.Identity.Client/Internal/Broker/BrokerInteractiveRequestComponent.cs

```diff
@@ -108,5 +117,40 @@ internal void ValidateResponseFromBroker(MsalTokenResponse msalTokenResponse)
                 throw new MsalServiceException(MsalError.BrokerResponseReturnedError, MsalErrorMessage.BrokerResponseReturnedError, null);
             }
         }
+
+        // Example auth code that shows that broker is required:
+        // msauth://wpj?username=joe@contoso.onmicrosoft.com&app_link=itms%3a%2f%2fitunes.apple.com%2fapp%2fazure-authenticator%2fid983156458%3fmt%3d8
+        public static bool IsBrokerRequiredAuthCode(string authCode, out string installationUri)
+        {
+            if (authCode.StartsWith(BrokerParameter.AuthCodePrefixForEmbeddedWebviewBrokerInstallRequired, StringComparison.OrdinalIgnoreCase))
+            //|| authCode.StartsWith(_serviceBundle.Config.RedirectUri, StringComparison.OrdinalIgnoreCase) // TODO: what is this?!
```

Comment:
@jennyf19 - still wasn't able to understand this piece of logic. Let's have a chat about different types of auth code uri that evo can send back.